src/lib/y2security/autoinst/lsm_config_reader.rb

```diff
@@ -0,0 +1,58 @@
+require "y2security/lsm/config"
+
+module Y2Security
+  module Autoinst
+    # This class is responsible of reading the LSM configuration declared in the AutoYaST profile
+    class LSMConfigReader
+      attr_reader :section
+
+      # Constructor
+      #
+      # @param section [AutoinstProfile::LSM::LsmSection]
+      def initialize(section)
+        @section = section
+      end
+
+      # Reads the LSM configuration defined in the profile modifying it accordingly
+      def read
+        return unless section
+
+        config.configurable = section.configurable
+        config.select(section.select) if section.select
+
+        configure_selinux(section.selinux) if section.selinux
+        configure_apparmor(section.apparmor) if section.apparmor
+      end
+
+    private
+
+      def configure_selinux(section)
+        selinux = config.selinux
+        return unless selinux
+
+        assign(selinux, section, :mode)
+        assign(selinux, section, :configurable)
+        assign(selinux, section, :selectable)
+        assign(selinux, section, :patterns)
+      end
+
+      def configure_apparmor(section)
+        apparmor = config.apparmor
+        return unless apparmor
+
+        assign(apparmor, section, :configurable)
+        assign(apparmor, section, :selectable)
+        assign(apparmor, section, :patterns)
+      end
+
+      def assign(mod, section, var)
```

Comment:
NP: having `section` as an _attr_reader_, you can omit the section param here. Can't you?